### ISSUE TITLE
Close context menu when refreshing logs

### ISSUE THREAD

**Issue by lindsve** [equinor/witsml-explorer#293]
**Is your feature request related to a problem? Please describe.**
When right clicking a log and clicking "Refresh", the context menu stays open until the updated list of logCurveInfos are fetched.

The same issue applies to other operations executed from the context menu as well. Can't remember exactly from the code, but it would be ideal to have a generalised way of handling this.

**Describe the solution you'd like**
Close the context menu as soon as the action for refreshing the log is sent

**Comment by HakonHesselberg** [comment]
I have created a fix for this. However, this is also an issue for all other selections in the context menu. Should we change it for those options too?

**Comment by lindsve** [comment]
Sounds reasonable to me. Once any selection is done, the context menu should be closed. I'm pretty sure this is the way it worked earlier as well, but somehow it does not now

**Comment by HakonHesselberg** [comment]
It also depends on the connection speed. If you get a server response almost immediately, you likely won't notice the delay. However, the slower the connection, the more noticeable.

If we want to do this for all context menus, could you expand the issue to include this?

**Comment by lindsve** [comment]
> It also depends on the connection speed. If you get a server response almost immediately, you likely won't notice the delay. However, the slower the connection, the more noticeable.

Correct. But the user interface should not be dependant on how long the underlying operation takes. This is out of scope for this issue, but a better way would be to have some kind of indication of background jobs running or data fetching (like a spinner)

> 
> If we want to do this for all context menus, could you expand the issue to include this?

Sure, I'll update the issue :)

**Comment by marmid74** [comment]
@lindsve , @HakonHesselberg  - Is this issue covered by another fix, or is this issue still relevant?

**Comment by HakonHesselberg** [comment]
The issue is still relevant and not yet solved.
